File: Backend/TranscomaAPI/Comun/Entidades/Cotizacion.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TranscomaAPI.Comun.Entidades
{
    public class Cotizacion: Entidad
    {
        private float _flete;
        private float _recargo;
        private float _gastosAdmin;
        private string _fechaSalida;
        private float _duracion;
        private float _total;

        public Cotizacion() { }

        public Cotizacion(float flete, float recargo, float gastosAdmin, string fechaSalida, float duracion, float total)
        {
            _flete = flete;
            _recargo = recargo;
            _gastosAdmin = gastosAdmin;
            _fechaSalida = fechaSalida;
            _duracion = duracion;
            _total = total;
        }

        public float Flete { get => _flete; set => _flete = value; }
        public float Recargo { get => _recargo; set => _recargo = value; }
        public float GastosAdmin { get => _gastosAdmin; set => _gastosAdmin = value; }
        public string FechaSalida { get => _fechaSalida; set => _fechaSalida = value; }
        public float Duracion { get => _duracion; set => _duracion = value; }
        public float Total { get => _total; set => _total = value; }
    }
}

File: frontend/src/app/models/Cotizacion.ts
import { EntidadBase } from "./EntidadBase";

export class Cotizacion extends EntidadBase {

    private _flete: number;
    private _recargo: number ;
    private _gastosAdmin: number ;
    private _fechaSalida: string ;
    private _duracion: number;
    private _total: number;

    constructor(data?: any) {
        super();
        if (data !== undefined) {
            this.$id = data.id;
            this.$Flete = data.flete;
            this.$Recargo = data.recargo;
            this.$GastosAdmin = data.gastosAdmin;
            this.$FechaSalida = data.fechaSalida;
            this.$Duracion = data.duracion;
            this.$Total = data.total;
        }
    }

    /** Getters y setters */
    public get $Flete(): number {
        return this._flete;
    }

    public set $Flete(value: number) {
        this._flete = value;
    }

    public get $Recargo(): number{
        return this._recargo;
    }

    public set $Recargo(value: number) {
        this._recargo = value;
    }

    public get $GastosAdmin(): number {
        return this._gastosAdmin;
    }

    public set $GastosAdmin(value: number) {
        this._gastosAdmin = value;
    }

    public get $FechaSalida(): string {
        return this._fechaSalida;
    }

    public set $FechaSalida(value: string) {
        this._fechaSalida = value;
    }

    public get $Duracion(): number {
        return this._duracion;
    }

    public set $Duracion(value: number) {
        this._duracion = value;
    }

    public get $Total(): number {
        return this._total;
    }

    public set $Total(value: number) {
        this._total = value;
    }
}
